evm/src/coverage/mod.rs

```diff
@@ -0,0 +1,346 @@
+mod visitor;
+pub use visitor::Visitor;
+
+use ethers::{
+    prelude::{sourcemap::SourceMap, sources::VersionedSourceFile},
+    types::Address,
+};
+use semver::Version;
+use std::{
+    collections::{BTreeMap, HashMap},
+    fmt::Display,
+    ops::AddAssign,
+    path::PathBuf,
+    usize,
+};
+
+/// A coverage map.
+///
+/// The coverage map collects coverage items for sources. It also converts hit data from
+/// [HitMap]s into their appropriate coverage items.
+///
+/// You **MUST** add all the sources before you start adding hit data.
+#[derive(Default, Debug, Clone)]
+pub struct CoverageMap {
+    /// A map of `(version, source id)` -> `source file`
+    sources: BTreeMap<(Version, u32), SourceFile>,
```

Comment:
General question: Is there an advantage to having this sorted compared to HashMap

Reply:
Mmm, probably not, no